xilem/examples/calc.rs

```diff
@@ -0,0 +1,326 @@
+        }
+        self.current_num_index = 0;
+        let num1 = self.numbers[0].parse::<f64>();
+        let num2 = self.numbers[1].parse::<f64>();
+        // Display format error or display the result of the operation.
+        self.result = Some(match (num1, num2) {
+            (Ok(num1), Ok(num2)) => self.operation.unwrap().perform_op(num1, num2).to_string(),
+            (Err(err), _) => err.to_string(),
+            (_, Err(err)) => err.to_string(),
+        });
+    }
+
+    fn on_delete(&mut self) {
+        if self.result.is_some() {
+            // Delete does not do anything with the result.
+            play_bell_sound();
+            return;
+        }
+        let mut num = self.get_current_number();
+        if !num.is_empty() {
+            num.remove(num.len() - 1);
+            self.set_current_number(num);
+        } else {
+            play_bell_sound();
+        }
+    }
+
+    fn negate(&mut self) {
+        // If there is a result, negate that after clearing and moving it to the first number
+        if self.result.is_some() {
+            self.move_result_to_left();
+        }
+        let mut num = self.get_current_number();
+        if num.is_empty() {
+            play_bell_sound();
+            return;
+        }
+        if num.starts_with('-') {
+            num.remove(0);
+        } else {
+            num = format!("-{num}");
+        }
+        self.set_current_number(num);
+    }
+}
+
+const DISPLAY_FONT_SIZE: f32 = 30.;
+const GRID_GAP: f64 = 2.;
+fn app_logic(data: &mut AppData) -> impl WidgetView<AppData> {
+    flex((
+        // Display
+        flex((
+            FlexSpacer::Flex(0.1),
+            label(data.numbers[0].clone()).text_size(DISPLAY_FONT_SIZE),
+            data.operation
+                .map(|operation| label(operation.as_str()).text_size(DISPLAY_FONT_SIZE)),
+            label(data.numbers[1].clone()).text_size(DISPLAY_FONT_SIZE),
+            data.result
+                .clone()
+                .map(|_| label("=").text_size(DISPLAY_FONT_SIZE)),
+            data.result
+                .clone()
+                .map(|result| label(result).text_size(DISPLAY_FONT_SIZE)),
+            FlexSpacer::Flex(0.1),
+        ))
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Center)
+        .main_axis_alignment(MainAxisAlignment::Start)
+        .gap(5.)
+        .flex(1.0),
+        FlexSpacer::Fixed(10.0),
+        // Top row
+        flex((
+            sized_box(button("CE", |data: &mut AppData| data.clear_entry()))
+                .expand()
+                .flex(1.0),
+            sized_box(button("C", |data: &mut AppData| data.clear_all()))
+                .expand()
+                .flex(1.0),
+            sized_box(button("DEL", |data: &mut AppData| data.on_delete()))
+                .expand()
+                .flex(1.0),
+            operator_button(MathOperator::Divide).flex(1.0),
+        ))
+        .direction(Axis::Horizontal)
```

Comment:
Yeah the issues of having a view sequence as the only function argument, while the modifiers of the widget via the builder pattern, I'm also not sure what to do here. Maybe promote smaller components as children, so that this is more obvious?

we could also think about having something like `flex(params: impl IntoFlexParams, children: impl ViewSequence)` where `()` (also) implements `IntoFlexParams`, but I'm not sure, this could look awkward as well (and may have inferior DX, as writing `.` after `flex()` gives the suggestions what can be configured, and allows more fancy modifications of the component, whether want that or not).